### PR TITLE
AAP-12257: Created and restructured AAP 2.4 release notes

### PR DESCRIPTION
**Pull request by sayjadha**
JIRA:

- Restructuring release notes: https://issues.redhat.com/browse/AAP-12257
- Creating AAP 2.4 release notes: https://issues.redhat.com/browse/AAP-12511

Following are the changes:

- Restructured AAP 2.4 release notes based on AAP versions instead of release date.
- Created AAP 2.4 release notes

Following are the files that warrant doc peer review:

- aap-24.adoc
- controller-440.adoc
- docs-24.adoc
- eda-24.adoc
- hub-464.adoc
- operator-240.adoc

Note: I'm working with Aine for the restructured files review.

### CODE REVIEW

File: downstream/titles/release-notes/topics/aap-24.adoc [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+* Fixed an issue where upgrading the Ansible Automation Platform did not update the nginx package by default.
+
+* Fixed an issue where an *awx* user was created without creating an *awx* group on execution nodes.
+
+* Fixed the assignment of package version variable to work with flat file inventories. 
+
+* Added a FQDN check for the automation hub hostname required to run the Skopeo commands.
+
+*  Fixed an issue such that the front end URL for Red Hat Single Sign On (SSO) is now properly configured after you specify the `sso_redirect_host` variable.
```
Comment (ianf77) [inline downstream/titles/release-notes/topics/aap-24.adoc:87]: Extra space.

File: downstream/titles/release-notes/topics/aap-24.adoc [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+* Fixed an issue when restoring a backup to an automation hub with a different hostname.
+
+== Technology Preview
+
+Some features in this release are currently classified as Technology Preview. Technology Preview features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process. Please note that Red Hat does not recommend using technology preview features for production, and Red Hat SLAs do not support technology preview functions.
+
+Following are the technology preview features: 
+
+* Ansible Automation Platform 2.4 adds the ability to install the automation controller for IBM Power (ppc64le), IBM Z (s390x), and IBM(R) LinuxONE (s390x) architectures as a Technology Preview feature.
```
Comment (ianf77) [inline downstream/titles/release-notes/topics/aap-24.adoc:101]: Is it necessary in a Technology Preview section, under a heading that says the following are technology preview features, to say that something is a technology preview feature?

File: downstream/titles/release-notes/topics/controller-41.adoc [hunk 3]
```diff
@@ -8,14 +10,14 @@ The name change reflects these enhancements and the overall position within the
...
-.New features
+.Automation controller 4.1 new features
 
 * Separate control and execution layers.
 * Automation execution environments are container images that provide a standardized way to build and distribute environments where automation runs.
 * Automation mesh allows distributed execution across on-premises environments, the hybrid cloud, and the edge. It replaces Ansible Tower and isolated nodes.
 
 
-.Removed
+.Automation controller 4.1 removed features
```
Comment (ianf77) [inline downstream/titles/release-notes/topics/controller-41.adoc:20]: Should  this be 'Deprecated and removed features' to be consistent with the previous section.

File: downstream/titles/release-notes/topics/controller-41.adoc [hunk 4]
```diff
@@ -8,14 +10,14 @@ The name change reflects these enhancements and the overall position within the
 Automation controller provides a standardized way to define, operate and delegate automation across the enterprise. It also introduces new, exciting technologies and an enhanced architecture that enables automation teams to scale and deliver automation rapidly to meet ever-growing business demand.
 
 
-.New features
+.Automation controller 4.1 new features
```
Comment (ianf77) [inline downstream/titles/release-notes/topics/controller-41.adoc:13]: Should this be 'New features and Enhancements' to be consistent with the previous section.

File: downstream/titles/release-notes/topics/docs-24.adoc [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+[[docs-2.4-intro]]
+= Ansible Automation Platform Documentation
+
+The documentation set for Red Hat Ansible Automation Platform 2.4 has had significant updates to improve the experience for our customers and the Ansible community. 
+
+
+.Enhancements
+
+* With the removal of the on-premise component Automation Services Catalog from Ansible Automation Platform 2.4 onwards, the Automation Services Catalog documentation too is removed from the Ansible Automation Platform 2.4 documentation.
```
Comment (ianf77) [inline downstream/titles/release-notes/topics/docs-24.adoc:11]: , all Automation Services catalog documentation is removed...

File: downstream/titles/release-notes/topics/docs-24.adoc [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+* The following documents are created to help you install and use Event-Driven Ansible, the newest capability of Ansible Automation Platform:
+
+** https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/2.4/html/getting_started_with_event_driven_ansible/index[Getting Started with Event-Driven Ansible]
+
+** https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/2.4/html/event_driven_ansible_controller_user_guide/index[Event-Driven Ansible User Guide ]
+
+In addition, sections of the https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/2.3/html/red_hat_ansible_automation_platform_planning_guide/index[Ansible Automation Platform Planning Guide]  
+and the https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/2.4/html/red_hat_ansible_automation_platform_installation_guide/index[Ansible Automation Platform Installation Guide]  are updated to include instructions for planning and installing Event-Driven Ansible.
```
Comment (ianf77) [inline downstream/titles/release-notes/topics/docs-24.adoc:20]: extra space before 'are updated'

File: downstream/titles/release-notes/topics/operator-240.adoc [hunk 7]
```diff
@@ -0,0 +1,30 @@
+// This is the release notes for Automation Platform Operator 2.4, the version number is removed from the topic title as part of the release notes restructuring efforts.
+
+[[operator-240-intro]]
+= Automation Platform Operator
+
+Ansible Automation Platform Operator provides cloud-native, push-button deployment of new Ansible Automation Platform instances in your OpenShift environment.
+
+.Enhancements
+
+* Starting with Ansible Automation Platform 2.4, the Platform Resource Operator can be used to create the following resources in Controller by applying YAML to your Openshift cluster:
```
Comment (ianf77) [inline downstream/titles/release-notes/topics/operator-240.adoc:10]: Should that be 'automation controller'?

File: downstream/titles/release-notes/topics/aap-24.adoc [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+* Added an option to configure SSO logout URL for automation hub.
+
+* Updated the ansible-lint RPM package to version 6.14.3.
+
+* Updated django for potential denial-of-service vulnerability in file uploads (CVE-2023-24580).
+
+* Updated sqlparse for ReDOS vulnerability (CVE-2023-30608).
+
+* Updated django for potential denial-of-service via Accept-Language headers (CVE-2023-23969).
```
Comment (ianf77) [inline downstream/titles/release-notes/topics/aap-24.adoc:44]: Why is this one 'via' while line 40 is 'in'?

**Review by ianf77 — COMMENTED** [review]
A few comments.

File: downstream/titles/release-notes/topics/eda-24.adoc [hunk 9]
```diff
@@ -0,0 +1,61 @@
+// This is the release notes for Event-Driven Ansible for AAP 2.4 release, the version number is removed from the topic title as part of the release notes restructuring efforts.
+
+[[eda-24-intro]]
+= Event-Driven Ansible
+
+Event-Driven Ansible is the newest capability of Ansible Automation Platform that is designed to enable automated response with user-defined, rules-based workflows. Event-Driven Ansible works by receiving events from third party tools, deciding on the actions to take and acting automatically.
```
Comment (jself-sudoku) [inline downstream/titles/release-notes/topics/eda-24.adoc:6]:
```suggestion
Event-Driven Ansible is the newest capability of Ansible Automation Platform that is designed to enable automated response with user-defined, rules-based workflows. Event-Driven Ansible works by receiving events from third party tools, deciding on the actions to take, and acting automatically.  
```
IBM style guide (https://www.ibm.com/docs/en/ibm-style?topic=punctuation-commas#commas-between-items-in-a-series)

File: downstream/titles/release-notes/topics/aap-24.adoc [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+
+= Ansible Automation Platform 2.4
+
+Red Hat Ansible Automation Platform simplifies the development and operation of automation workloads for managing enterprise application infrastructure lifecycles. It works across multiple IT domains including operations, networking, security, and development, as well as across diverse hybrid environments. Simple to adopt, use, and understand, Red Hat Ansible Automation Platform provides the tools needed to rapidly implement enterprise-wide automation, no matter where you are in your automation journey.
+
+== New features and enhancements
+
+This release of Ansible Automation Platform features the following enhancements:
+
+* Previously, the Execution Environment container images were based on RHEL 8 only. With Ansible Automation Platform 2.4 onwards, the Execution Environment container images are now available on RHEL 9 too.
```
[the remark this answers is not in the document]
Comment (sayjadha) [reply downstream/titles/release-notes/topics/aap-24.adoc:14]: Made the change. Thank you.

File: downstream/titles/release-notes/topics/aap-24.adoc [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+
+* Smart inventories have been deprecated in favor of constructed inventories and will be removed in a future release.
+
+== Bug fixes
+
+The following bugs were fixed in this release of Ansible Automation Platform:
+
+* Installer now ensures that collection auto signing cannot be enabled without enabling the collection signing service.
+
+*  Fixed an issue with restoring backups when the installed automation controller version is different from the backup version.
```
[the remark this answers is not in the document]
Comment (sayjadha) [reply downstream/titles/release-notes/topics/aap-24.adoc:69]: Good catch, corrected the spacing.

File: downstream/titles/release-notes/topics/aap-24.adoc (hunk 1 above)
Comment (sayjadha) [reply downstream/titles/release-notes/topics/aap-24.adoc:87]: Corrected the spacing here as well, thank you.

File: downstream/titles/release-notes/topics/eda-24.adoc [hunk 12]
```diff
@@ -0,0 +1,61 @@
+// This is the release notes for Event-Driven Ansible for AAP 2.4 release, the version number is removed from the topic title as part of the release notes restructuring efforts.
+
+[[eda-24-intro]]
+= Event-Driven Ansible
+
+Event-Driven Ansible is the newest capability of Ansible Automation Platform that is designed to enable automated response with user-defined, rules-based workflows. Event-Driven Ansible works by receiving events from third party tools, deciding on the actions to take and acting automatically.  
+
+Event-Driven Ansible is included in Ansible Automation Platform, making the platform even more capable as a single enterprise automation solution.  Using Event-Driven Ansible, domain experts can easily create end-to-end fully automated OpsAsCode scenarios for a  broad array of use cases across the IT landscape.  By eliminating high-volume routine tasks and automatically responding to changing conditions, teams are free to innovate more, sleep more, and act consistently and accurately at scale.
```
Comment (jself-sudoku) [inline downstream/titles/release-notes/topics/eda-24.adoc:8]:
```suggestion
Event-Driven Ansible is included in Ansible Automation Platform, making the platform even more capable as a single enterprise automation solution. Using Event-Driven Ansible, domain experts can easily create end-to-end, fully automated OpsAsCode scenarios for a broad array of use cases across the IT landscape. By eliminating high-volume routine tasks and automatically responding to changing conditions, teams are free to innovate more and act consistently and accurately at scale.
```
Just removed a couple random extra spaces. Suggestion to remove "sleep more". Did this come from some of the marketing content?

File: downstream/titles/release-notes/topics/aap-24.adoc (hunk 2 above)
Comment (sayjadha) [reply downstream/titles/release-notes/topics/aap-24.adoc:101]: Agree, this is repetitive. Removed the 'as a tech. preview feature' text.

File: downstream/titles/release-notes/topics/controller-41.adoc (hunk 4 above)
Comment (sayjadha) [reply downstream/titles/release-notes/topics/controller-41.adoc:13]: Made the change, thank you.

File: downstream/titles/release-notes/topics/docs-24.adoc (hunk 5 above)
Comment (sayjadha) [reply downstream/titles/release-notes/topics/docs-24.adoc:11]: Made the change.

File: downstream/titles/release-notes/topics/docs-24.adoc (hunk 6 above)
Comment (sayjadha) [reply downstream/titles/release-notes/topics/docs-24.adoc:20]: Removed the extra spacing, thanks

File: downstream/titles/release-notes/topics/aap-24.adoc (hunk 8 above)
Comment (sayjadha) [reply downstream/titles/release-notes/topics/aap-24.adoc:44]: Good point, changed it to 'in'

File: downstream/titles/release-notes/topics/eda-24.adoc (hunk 12 above)
Comment (sayjadha) [reply downstream/titles/release-notes/topics/eda-24.adoc:8]: Thank you, removed the extra spaces. Yes, the info. was shared by Joe Pisciotta, but now that I'm re-reading it, it does sound off. I'm confirming if we can remove these specific words.

[the remark this answers is not in the document]
Comment (sayjadha) [reply downstream/titles/release-notes/topics/eda-24.adoc:8]: Yes, the info. was shared by Joe Pisciotta, but now that I'm re-reading it, it does sound off. I'm confirming if we can remove these specific words.

[the remark this answers is not in the document]
Comment (sayjadha) [reply downstream/titles/release-notes/topics/eda-24.adoc:8]: Confirmed with PM and changed it to 'innovate more efficiently, and act consistently...". Thank you for catching that!

Comment (sayjadha) [reply downstream/titles/release-notes/topics/eda-24.adoc:8]: Confirmed with PM and changed it to 'innovate more efficiently, and act consistently...". Thank you for catching that!

**Comment by sayjadha** [comment]
@ariordan-redhat : Thank you for the structural review. @jself-sudoku , @ianf77 : Thank you for the peer review. Your feedback is incorporated. Request recheck and squash-&-merge if everything looks good.

**Comment by jself-sudoku** [comment]
LGTM! Merging.

**Pull request merged**